### DISCUSSION TITLE
Tab item foreground

### DISCUSSION THREAD

**Discussion by elveejay1** [AvaloniaUI/Avalonia#8701] [Q&A]
I have a TabControl with a number of TabItems in it.
How do I change the foreground/text color?
How do I make the foreground/text color change on
* MouseOver
* Selected

I have experimented a little with

```xaml
	<Style Selector="TabItem:pointerover /template/ ContentPresenter#PART_ContentPresenter">
			<Setter Property="Background" Value="Transparent" />
		</Style>
		<Style Selector="TabItem:pointerover /template/ ContentPresenter#PART_ContentPresenter">
			<Setter Property="TextBlock.Foreground" Value="Red"/>
		</Style>
		<Style Selector="TabItem:pointerover">
			<Setter Property="Foreground" Value="Red"/>
			<Setter Property="TextBlock.Foreground" Value="Red"/>
		</Style>
		<Style Selector="TabItem:selected">
			<Setter Property="Foreground" Value="Purple"/>
			<Setter Property="Margin" Value="0 0 0 0"/>
			<Setter Property="Padding" Value="10 0"/>
		</Style>
```

but neither seems to work.

Sorry for the bad formatting - the forum doesn't always want to respect newlines

**Comment by timunie** [comment]
> Sorry for the bad formatting - the forum doesn't always want to respect newlines

You can, see: 

````
```xml  <- optional language tag
Your content goes here.
```
````

back to your topic: 

You can inspect the original style here: https://github.com/AvaloniaUI/Avalonia/blob/765c4d7094b054cdd3d21d73964258b6829ae01d/src/Avalonia.Themes.Fluent/Controls/TabItem.xaml#L70-L80 and normally this should be overrideable

**Reply by elveejay1** [reply]
I tried that, and as before, only the Background seemed to work.

**Reply by maxkatz6** [reply]
If you copy these styles and modify, it will work fine. If you want to write some more advanced or custom styles, you can visit this page https://docs.avaloniaui.net/docs/styling/troubleshooting

**Reply by elveejay1** [reply]
I now have three different style attempts:

```xml
		<Style Selector="TabItem:pointerover /template/ ContentPresenter#PART_ContentPresenter">
			<Setter Property="Background" Value="Transparent" />
			<Setter Property="TextBlock.Foreground" Value="Red"/>
		</Style>
				<Style Selector="TabItem:pointerover">
			<Setter Property="Foreground" Value="Red"/>
			<Setter Property="TextBlock.Foreground" Value="Red"/>
		</Style>

		<Style Selector="TabItem:pointerover /template/ Border#PART_LayoutRoot">
			<Setter Property="Background" Value="Red" />
			<Setter Property="TextBlock.Foreground" Value="Green" />
		</Style>
```

I can change the background, but not the text foreground color.

The styles have been added in my UserControl.Styles, while the TabItem resides in a TabControl in a Grid inside the UserControl.
There are no further Styles that could  break my styling above.

The TabItems all look like this:

```xml
			<TabItem Header="Setup" FontSize="25" Height="100" Width="250" BorderThickness="3" Background="White">
				<!-- content -->
			</TabItem>
```

This is super strange...

My pointerover-background is Red, as expected.

**Reply by maxkatz6** [reply]
Works for me
![image](https://user-images.githubusercontent.com/3163374/183590983-86fca77f-5cf2-4781-982d-6cf607dbbac6.png)

**Reply by maxkatz6** [reply]
[AvaloniaApplication1.zip](https://github.com/AvaloniaUI/Avalonia/files/9288213/AvaloniaApplication1.zip)

**Reply by maxkatz6** [reply]
You need to try to isolate your problem. Also use DevTools to see what's property's value and priority of the value.

**Reply by elveejay1** [reply]
Thanks. It is super strange - it is the exact same code we have for the style...

I assume (I'm very new to Avalonia - started last Friday), that if I have two stylings that affect the same object, that the last one always wins? Or can a style otherwise be configured to be more important than another?

**Reply by timunie** [reply]
Are you probably using another Theme than Avalonia.Fluent? Like Material Avalonia or similar? If so, look up those Styles.

**Reply by timunie** [reply]
> that the last one always wins

Yes, the style that is closest to your TabControl will win.

**Reply by elveejay1** [reply]
In App.xaml I have

```xml
    <Application.Styles>
        <FluentTheme Mode="Light"/>
		<StyleInclude Source="avares://Material.Icons.Avalonia/App.xaml"></StyleInclude>

	</Application.Styles>
```

In my MainWindow, I have a StyleInclude, and in the file that I include, TextBlock has a style with a Setter which sets the Foreground. Disabling that made the TextBlock in the TabItem work. This has so far been a huge frustration of mine - that I can't stop styles from one control to affect those controls when they are part of another control. For instance I found it super annoying, that when I style a TextBlock (at least its color), it also affects Buttons and now latest TabItems.

Just for my learning experince - is there a way to stop that?

**Reply by timunie** [reply]
I think there will be a way in the next release. For now, I'd use classes instead.

**Reply by elveejay1** [reply]
Thanks both of you. You've been super helpful :-)

**Reply by maxkatz6** [reply]
> TextBlock has a style with a Setter which sets the Foreground

If you do this, foreground property inheritance won't work. 
In styles above we set foreground on the content control, and this value is inherited to the child TextBlock.

**Reply by maxkatz6** [reply]
Foreground globally, you can set it on Window (locally or with style).
